src/Stratis.Bitcoin.Features.PoA/ConsensusRules/PoACoinviewRule.cs

```diff
@@ -9,13 +9,11 @@ namespace Stratis.Bitcoin.Features.PoA.ConsensusRules
 {
     public class PoACoinviewRule : CoinViewRule
     {
-        private PoANetwork network;
+        private readonly PoANetwork network;
 
-        public override void Initialize()
+        public PoACoinviewRule(PoANetwork network)
```

Comment:
hmm well the approach we had before worked, I think if we want to refactor how we do rules lets think about it first.

Reply:
Because it's experiential I've introduced it only for PoA rules